src/Flow.jl

```diff
@@ -85,9 +102,11 @@ struct Flow{D, T, Sf<:AbstractArray{T}, Vf<:AbstractArray{T}, Tf<:AbstractArray{
     U :: NTuple{D, T} # domain boundary values
     Δt:: Vector{T} # time step (stored in CPU memory)
     ν :: T # kinematic viscosity
+    g :: Function  # (possibly time-varying) body force field
+    tagBF :: Bool # whether to switch on the body force
```

Comment:
you do not really need `tagBF`. Consider de following workflow for multiple dispatch on `g`
```
foo(g, ::Val{nothing}) = "empty"
foo(g, ::Val) = "not empty"

g = nothing
foo(g, Val{g}()) # returns "empty"
g = x -> 0
foo(g, Val{g}()) # returns "not empty"
```

Reply:
You only need `foo(::Nothing)="nothing"`.